src/types.ts

```diff
@@ -59,4 +51,21 @@ export type Config = {
   // Legacy but still supported
   host?: string;
   port?: number;
-};
\ No newline at end of file
+};
+
+export interface SeedConfig extends BaseConfig {
+  seed: unknown;
+  keys: never;
+};
+
+export interface KeysConfig extends BaseConfig {
+  seed: never;
+  // Cryptographic keys
+  keys: {
+    public: string;
+    private: string;
+    id: string;
+  };
+};
+
+export type Config = KeysConfig | SeedConfig;
```

Comment:
@staltz I am attempting to create a `Config` type that requires one OR the other of a property. This makes it's way through the typescript compiler without issue, but wondering if there is a better solution.